homeassistant/components/plex/config_flow.py

```diff
@@ -225,6 +234,45 @@ def __init__(self):
         _LOGGER.debug("Imported Plex configuration")
         return await self.async_step_server_validate(import_config)
 
+    async def async_step_plex_website_auth(self):
+        """Begin external auth flow on Plex website."""
+        self.hass.http.register_view(PlexAuthorizationCallbackView)
+        payload = {
+            "X-Plex-Device-Name": X_PLEX_DEVICE_NAME,
+            "X-Plex-Version": X_PLEX_VERSION,
+            "X-Plex-Product": X_PLEX_PRODUCT,
+            "X-Plex-Device": self.hass.config.location_name,
+            "X-Plex-Platform": X_PLEX_PLATFORM,
+            "X-Plex-Model": "Plex OAuth",
+        }
+        session = async_get_clientsession(self.hass)
+        self.plexauth = PlexAuth(payload, session)
+        await self.plexauth.initiate_auth()
+        forward_url = f"{self.hass.config.api.base_url}{AUTH_CALLBACK_PATH}?flow_id={self.flow_id}"
+        auth_url = await self.hass.async_add_executor_job(
```

Comment:
Looking at the library we don't seem to need the executor here. It's just creating a url, right?

Reply:
Yes, you're right. No I/O within this call, just string manipulation. Reverted.